nixos/modules/services/networking/wg-quick.nix

```diff
@@ -11,6 +11,16 @@ let
   interfaceOpts = { ... }: {
     options = {
 
+      type = mkOption {
+        example = "amneziawg";
+        default = "wireguard";
+        type = types.enum ["wireguard" "amneziawg"];
+        description = ''
+          The type of the interface. AmneziaWG is a censor-resistant fork of WireGuard.
+          You should reboot system to make amneziawg kernel module load if you use it.
+        '';
+      };
+
```

Comment:
It seems it would be required to add support for AmneziaWG specific parameters: `Jc`, `Jmin`, `Jmax`, `S1`, `S2`, `H1`, `H2`, `H3`, `H4`.